### DISCUSSION TITLE
Dynamic Routing from the root page

### DISCUSSION THREAD

**Discussion by useDeep** [vercel/next.js#51846] [App Router]
In the new app dir, the dynamic routing is achieved by the file and folder structure by `wrapping a folder's name in square brackets: [folderName]. For example, [id] or [slug]` in a new folder, name same as the route name.

Ex: app/blog/[slug]/page.js 
URL Ex: **www .example/blog/3/**

But what should be done to dynamically route from the first level i.e., from the root page?

URL Ex: **www .example/3**

**Comment by hesalx** [comment]
`app/[id]/page.jsx` worked just fine.

`http://localhost:3000/3`
```js
export default function Page(args) {
  return <>{JSON.stringify(args)}</> // {"params":{"id":"3"},"searchParams":{}}
}
```

Any constant path like `app/about/page.jsx` takes priority over a dynamic one, so it is fine to have a dynamic segment at the root.

**Reply by useDeep** [reply]
Thanks! It worked :)
